provider-example/src/sign/ecdsa.rs

```diff
@@ -0,0 +1,58 @@
+use pkcs8::DecodePrivateKey;
+use pki_types::PrivateKeyDer;
+use rustls::{
+    sign::{Signer, SigningKey},
+    SignatureAlgorithm, SignatureScheme,
+};
+use signature::RandomizedSigner;
+use signature::SignatureEncoding;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct EcdsaSigningKeyP256 {
+    key: Arc<p256::ecdsa::SigningKey>,
+    scheme: SignatureScheme,
+}
+
+impl TryFrom<PrivateKeyDer<'_>> for EcdsaSigningKeyP256 {
+    type Error = pkcs8::Error;
+
+    fn try_from(value: PrivateKeyDer<'_>) -> Result<Self, Self::Error> {
+        match value {
+            PrivateKeyDer::Pkcs8(der) => {
+                p256::ecdsa::SigningKey::from_pkcs8_der(der.secret_pkcs8_der()).map(|kp| Self {
```

Comment:
unnecessary qualification:
```suggestion
                ecdsa::SigningKey::from_pkcs8_der(der.secret_pkcs8_der()).map(|kp| Self {
```

Reply:
I think this is necessary: `p256` is the name of a crate so this refers to: https://docs.rs/p256/latest/p256/ecdsa/type.SigningKey.html

Reply:
Hm, it seems to compile/run when I remove it but it's such a tiny nit pick I don't think it matters much either way :laughing:

Reply:
Ah, yeah, same here. It seems `ecdsa` is the name of a crate full of generics, and `p256::ecdsa` is the same set of generics reified with the p256 curve.